src/media/class-wpml-beaver-builder-update-media-factory.php

```diff
@@ -2,13 +2,33 @@
 
 class WPML_Beaver_Builder_Update_Media_Factory implements IWPML_PB_Media_Update_Factory {
 
+	/** @var  */
+	private $media_translate;
+
 	public function create() {
 		global $sitepress;
 
 		return new WPML_Page_Builders_Update_Media(
 			new WPML_Page_Builders_Update( new WPML_Beaver_Builder_Data_Settings() ),
 			new WPML_Translation_Element_Factory( $sitepress ),
-			new WPML_Beaver_Builder_Media_Nodes_Iterator( new WPML_Beaver_Builder_Media_Node_Provider() )
+			new WPML_Beaver_Builder_Media_Nodes_Iterator(
+				new WPML_Beaver_Builder_Media_Node_Provider( $this->get_media_translate() )
+			),
+			new WPML_Page_Builders_Media_Usage( $this->get_media_translate(), new WPML_Media_Usage_Factory() )
 		);
 	}
+
+	/** @return WPML_Page_Builders_Media_Translate */
+	private function get_media_translate() {
+		global $sitepress;
+
+		if ( ! $this->media_translate ) {
```

Comment:
@strategio Do you really need to store this. From what I can this function will only ever be called once.

Reply:
@brucepearson, `get_media_translate` is called twice. I created this private method because we need to use the same instance of `WPML_Page_Builders_Media_Translate` in `WPML_Beaver_Builder_Media_Node_Provider` and `WPML_Page_Builders_Media_Usage`.

It's not really needed to keep a reference of the object, but I think having a private method to get the object will emphasize the need for a single instance.